app/services/person_constructor.rb

```diff
@@ -72,6 +73,12 @@ def strip_whitespace_from_values
     params.merge!(params){ |k, v, _| v.try(:strip) || v }
   end
 
+  def normalize_zip_code
+    if params[:zip_code]
+      params[:zip_code] = params[:zip_code].gsub!(/\A(\d{5})[^\w]?(\d{4})?\z/, '\1')
```

Comment:
Why not just trim it and capture the first 5 characters?  We're not keeping the later 4.  You could just do:
```ruby
params[:zip_code][0..4]
'555554444­'.slice(0.­.4)
#=> "55555"
'555554444­'.scan(/\A­\d{5}/)
#=> ["55555"]
```